bigquery/deployment/pom.xml

```diff
@@ -0,0 +1,50 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <parent>
+        <groupId>io.quarkiverse</groupId>
```

Comment:
I think I would prefer having a subgroup for each repository. Otherwise it's going to be a nightmare. Even for the Hibernate projects, we moved to using a subgroup for each project and this is going to be a lot bigger.

@aloubyansky WDYT?

Reply:
Let's clarify why that would be a nightmare. You want to be able to filter things based on groupIds?

Reply:
You will have a big pile of artifacts in a directory rather than having each project properly separated in a directory. No idea from which project an artifact is.

If we have a subgroup, we will have each project at the root of the groupId and then in a subdirectory all the artifacts of the project.

Also you will entirely avoid conflicts this way, which is nice.

Reply:
Ok, that's reasonable.

Reply:
Usually, project are compound names and artifact ID are coma separated terms.
So in this case should I use `io.quarkiverse.google.cloud.services` as a group ID (OK, I can use gcp as a single name but I give this as an example).
Looking at what OpenTracing contrib are using, they all use the same groupId that's why we do it this way.

The __folder__ issue doesn't bother me, nobody really browse Maven central as a tree of folder anyway.

The naming conflict is a possible issue. For example another repository may offer a google-cloud-services extension for something else ... This is avoidable with carefull review, anyway when a repository will be created in the org there will be some sort of "review process" to be integrated in the quarkiverse (git org, maven repo, BOM maybe ?).

Anyway, I'm OK for both :)
Just giving some counter arguments.

Reply:
Well, the fact that it doesn't bother YOU is not enough. Question is: will it be annoying for some users? And I'm sure it will be.

Having a ton of random unsorted artifacts at different versions (releases won't be synced) at the root of the groupId is not good. We could have hundreds of artifacts there. Your project is a good example: you have several artifacts in there and that will be the case for most projects.

I would go with:
`io.quarkiverse.googlecloudservices` as both the groupId and the root package.

Having a specific root for both will avoid conflicts and split packages and make things easier if we need to script things.

But let's wait for @aloubyansky 's view on this before changing things.

Reply:
For me the important point is that the release cycles of all these projects will be totally independent.

Reply:
And I think it should be a process when someone wants to add something to quarkiverse.

We affect him a root that is descriptive and unique that should be used for both the groupId and the root package.

Reply:
For root package, for the moment I don't update is so it's still `io.quarkus`.
Max said that we also need the capability to easily move an extension from the Quarkiverse to Quarkus if needed.
Using a different package name could be an issue.
On the other hand, I agree that having completely separate namespaces (groupId and root package) could be easier to avoid namming conflict if a lot of extensions ends up in the Quarkiverse.

So, I'm OK for using separate groupId and packages, just tell me ;)

Reply:
Yeah, but I don't think we want everyone to use the `io.quarkus` package.

And we could have conflicts between Quarkiverse and Core.

So I really don't think using `io.quarkus` is a good idea.

Reply:
I agree with the `io.quarkiverse.` prefix followed by a project-specific part. In general, I don't see why not allow dots in that project-specific part. To me `io.quarkiverse.google.cloud.services` isn't looking bad. It's even easier to read than a single word.

Reply:
No it's a bad idea to use dots. Dots are used as a structural separator. I'm pretty sure you will end up with `io/quarkiverse/google/cloud/services` on Maven Central and that's definitely not what we want.

I agree it would be better with dashes but unfortunately you can't use them in package names.

So I would use what I suggested given the constraints we have for package names.

Reply:
Of course it will be subdirs but why is that bad?

Reply:
Because what you want is a structure such as:
```
io/quarkiverse/<project>/
```

Reply:
Trust me on this, if we don't have a clean structure, it's going to be a major pain in the future when we have 100 projects.

Reply:
I agree with @gsmet on this one, groupId is hierarchical, if I use `io.quarkiverse.google.cloud.services` it means it's a services project on a cloud projects on a google projet on the Quarkiverse.

This feels a bit wrong as it's just a GoogleCloudServices project on the Quarkiverse.

Reply:
Ok, then we all agree.

Reply:
- *groupId*: `io.quarkiverse.googlecloudservices` or I can use `io.quarkiverse.gcp` as gcp stands for Google Cloud Platform ...
- *artifactId*: `google-cloud-services`
- *root package*: same as groupId

Reply:
I'd let you pick @loicmathieu the final groupId. The rest looks good to me. Thanks!

Reply:
@gsmet @aloubyansky I rename everything is it OK now ?